flink/v2.0/flink/src/main/java/org/apache/iceberg/flink/sink/dynamic/TableUpdater.java

```diff
@@ -90,6 +109,53 @@ private void findOrCreateTable(TableIdentifier identifier, Schema schema, Partit
     }
   }
 
+  private void updateTablePropertiesIfNeeded(TableIdentifier identifier) {
+    if (tablePropertiesUpdater == null) {
+      return;
+    }
+
+    Map<String, String> currentProperties = cache.properties(identifier);
+    Map<String, String> updatedProperties =
+        tablePropertiesUpdater.apply(identifier.toString(), currentProperties);
+
+    if (updatedProperties == null || Objects.equals(currentProperties, updatedProperties)) {
+      return;
+    }
+
+    LOG.info(
+        "Updating table {} properties from {} to {}",
+        identifier,
+        currentProperties,
+        updatedProperties);
+    Table table = catalog.loadTable(identifier);
+    try {
+      UpdateProperties updateApi = table.updateProperties();
+
+      // Remove properties that are no longer present
```

Comment:
So we consider Flink job as source of truth for properties and remove properties on tables , if added from external pipelines ?

In large scale ingestion pipelines, if we need to add a new property, we will need to then update the TablePropertiesUpdaterImpl and redeploy right ? This may not be a feasible as it may also put too much pressure on catalog on startup.